### PR TITLE
DBZ-2773 Trigger non-core connectors to resolve cross-repo builds

### PR DESCRIPTION
**Pull request by ani-sha**
https://issues.redhat.com/browse/DBZ-2773

### CODE REVIEW

File: .github/workflows/incubator-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Build Incubator Workflow
+
+on:
+  push:
+    paths:
+      - 'debezium-core/**'
```
Comment (jpechane) [inline .github/workflows/incubator-workflow.yml:6]: `debezium-core` depends on `debezium-api` so it should be included here and in other instances too

File: .github/workflows/incubator-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          repository: debezium/debezium-incubator
+          path: incubator
+      - name: Cache Maven Packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Build Debezium Incubator
```
Comment (jpechane) [inline .github/workflows/incubator-workflow.yml:27]: You need to build core first here as you do for vitess workflow

**Comment by jpechane** [comment]
@ani-sha Could you please check the actions log? It seems to me that it checks out master branches instead of branch for which the action is part

**Comment by ani-sha** [comment]
> @ani-sha Could you please check the actions log? It seems to me that it checks out master branches instead of branch for which the action is part

@jpechane yes it's checking out to master.

**Comment by ani-sha** [comment]
@jpechane Added the changes! Now it checks out to the respective `branch` for core and `master` for `vitess`, `incubator`

**Comment by jpechane** [comment]
@ani-sha Applied, good job!

**Pull request merged**